src/Menu/PickerMenuBuilder.php

```diff
@@ -0,0 +1,199 @@
+     * @param EventDispatcherInterface $dispatcher
+     * @param FactoryInterface         $factory
+     * @param RendererInterface        $renderer
+     */
+    public function __construct(RequestStack $requestStack, RouterInterface $router, EventDispatcherInterface $dispatcher, FactoryInterface $factory, RendererInterface $renderer)
+    {
+        $this->requestStack = $requestStack;
+        $this->router = $router;
+        $this->dispatcher = $dispatcher;
+        $this->factory = $factory;
+        $this->renderer = $renderer;
+    }
+
+    /**
+     * Sets the page picker label.
+     *
+     * @param string $pagePickerLabel
+     */
+    public function setPagePickerLabel($pagePickerLabel)
+    {
+        $this->pagePickerLabel = $pagePickerLabel;
+    }
+
+    /**
+     * Sets the file picker label.
+     *
+     * @param string $filePickerLabel
+     */
+    public function setFilePickerLabel($filePickerLabel)
+    {
+        $this->filePickerLabel = $filePickerLabel;
+    }
+
+    /**
+     * Creates the file menu.
+     *
+     * @param BackendUser $user
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    public function createMenu(BackendUser $user)
+    {
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (null === $request) {
+            throw new \RuntimeException('No request object given');
+        }
+
+        $menu = $this->factory->createItem('picker');
+
+        if ($user->hasAccess('page', 'modules')) {
+            $this->addPagePickerItem($menu, $request);
+        }
+
+        if ($user->hasAccess('files', 'modules')) {
+            $this->addFilePickerItem($menu, $request);
+        }
+
+        $this->dispatcher->dispatch(
+            ContaoCoreEvents::BUILD_PICKER_MENU,
+            new BuildPickerMenuEvent($this->factory, $menu)
+        );
+
+        return $this->renderer->render($menu);
+    }
+
+    /**
+     * Adds the page picker item.
+     *
+     * @param ItemInterface $menu
+     * @param Request       $request
+     */
+    private function addPagePickerItem(ItemInterface $menu, Request $request)
+    {
+        $item = $this->factory->createItem(
+            $this->pagePickerLabel,
+            ['uri' => $this->route('contao_backend', 'page', $request)]
+        );
+
+        $item->setLinkAttribute('class', 'pagemounts');
+
+        if ('page' === $request->query->get('do')) {
+            $item->setAttribute('class', 'active');
+        }
+
+        $menu->addChild($item);
+    }
+
+    /**
+     * Adds the file picker item.
+     *
+     * @param ItemInterface $menu
+     * @param Request       $request
+     */
+    private function addFilePickerItem(ItemInterface $menu, Request $request)
+    {
+        $item = $this->factory->createItem(
+            $this->filePickerLabel,
+            ['uri' => $this->route('contao_backend', 'files', $request)]
+        );
+
+        $item->setLinkAttribute('class', 'filemounts');
```

Comment:
I suppose this comment relates to line 177, doesn't it? Because `setCurrent()` will not set the link attribute.